runners/gearpump/src/main/java/org/apache/beam/runners/gearpump/TestGearpumpRunner.java

```diff
@@ -52,7 +53,10 @@ public static TestGearpumpRunner fromOptions(PipelineOptions options) {
   @Override
   public GearpumpPipelineResult run(Pipeline pipeline) {
     GearpumpPipelineResult result = delegate.run(pipeline);
+    PipelineResult.State state = result.waitUntilFinish();
     cluster.stop();
+    assert(state == PipelineResult.State.DONE);
```

Comment:
I think here that you want to try to raise some underlying exception / give the user a message, or there are RunnnableOnService tests that won't pass. Or if I recall correctly, `waitUntilFinish` could throw. But I think that design decision has gone back and forth a couple of times.